File: 2022/18/18.py
with open("inputs/18.txt") as file:
    cubes = set()
    for line in file.read().splitlines():
        cubes.add(tuple(int(x) for x in line.split(",")))


def part_1():
    count = 0
    for x, y, z in cubes:
        count += int(((x + 1, y, z) not in cubes))
        count += int(((x - 1, y, z) not in cubes))
        count += int(((x, y + 1, z) not in cubes))
        count += int(((x, y - 1, z) not in cubes))
        count += int(((x, y, z + 1) not in cubes))
        count += int(((x, y, z - 1) not in cubes))
    return count


def part_2():
    pass


print(f"Part 1: {part_1()}")
print(f"Part 2: {part_2()}")

File: 2022/18/18.rb
# frozen_string_literal: true

require 'set'

path = File.expand_path('../inputs/18.txt', __dir__)
cubes = Set.new(File.readlines(path, chomp: true).map { _1.split(',').map(&:to_i) })

part1 = cubes.reduce(0) do |current_sum, (x, y, z)|
  current_sum += 1 unless cubes.include?([x + 1, y, z])
  current_sum += 1 unless cubes.include?([x - 1, y, z])
  current_sum += 1 unless cubes.include?([x, y + 1, z])
  current_sum += 1 unless cubes.include?([x, y - 1, z])
  current_sum += 1 unless cubes.include?([x, y, z + 1])
  current_sum += 1 unless cubes.include?([x, y, z - 1])
  current_sum
end

puts "part 1:  #{part1} "
